core/src/scene/scene.h

```diff
@@ -32,6 +33,10 @@ class Scene {
     Scene();
     ~Scene();
 
+    using ArrayValue = std::vector<double>;
+    using Value = variant<none_type, bool, double, std::string, ArrayValue>;
+    std::unordered_map<std::string, Value> m_globals;
```

Comment:
I'm not clear why this is a public member. If it *needs* to be a public member, it should not be named with `m_`, otherwise it should be private like the other members.